Facebook.AccountKit.iOS/source/Facebook.AccountKit/ApiDefinition.cs

```diff
@@ -0,0 +1,600 @@
+        [Export("setError:")]
+        void SetError(NSError error);
+
+        // @optional -(AKFTextPosition)textPositionForState:(AKFLoginFlowState)state;
+        [Export("textPositionForState:")]
+        TextPosition TextPositionForState(LoginFlowState state);
+
+        // @optional -(Theme * _Nullable)theme;
+        [NullAllowed, Export("theme")]
+        //[Verify(MethodToProperty)]
+        Theme Theme { get; }
+    }
+
+    interface IUIManager { }
+
+    // @protocol AKFAdvancedUIManager <AKFUIManager>
+    [Protocol, Model]
+    [BaseType(typeof(NSObject), Name = "AKFAdvancedUIManager")]
+    interface AdvancedUIManager : UIManager
+    {
+    }
+
+    // @protocol AKFAdvancedUIActionController <AKFActionController>
+    [Protocol, Model]
+    [BaseType(typeof(NSObject), Name = "AKFAdvancedUIActionController")]
+    interface AdvancedUIActionController : ActionController
+    {
+    }
+
+    // @protocol AKFUIManaging <NSObject>
+    [Protocol, Model]
+    [BaseType(typeof(NSObject), Name = "AKFUIManaging")]
+    interface UIManaging
+    {
+        // @required @property (nonatomic, strong) id<AKFUIManager> uiManager;
+        [Abstract]
+        [Export("uiManager", ArgumentSemantic.Strong)]
+        IUIManager UiManager { get; set; }
+
+        // @required -(void)setAdvancedUIManager:(id<AKFAdvancedUIManager>)uiManager;
+        [Abstract]
+        [Export("setAdvancedUIManager:")]
+        void SetAdvancedUIManager(AdvancedUIManager uiManager);
+
+        // @required -(void)setTheme:(Theme *)theme;
+        [Abstract]
+        [Export("setTheme:")]
+        void SetTheme(Theme theme);
+    }
+
+    interface IUIManaging { }
+
+    // @protocol AKFViewController <AKFUIManaging, AKFConfiguring>
+    [Protocol, Model]
+    [BaseType(typeof(NSObject), Name = "AKFViewController")]
+    interface ViewController : UIManaging, Configuring
+    {
+        //[Abstract]
+        [Wrap("WeakDelegate")]
+        ViewControllerDelegate Delegate { get; set; }
+
+        // @required @property (nonatomic, weak) id<AKFViewControllerDelegate> delegate;
+        [Abstract]
+        [NullAllowed, Export("delegate", ArgumentSemantic.Weak)]
+        NSObject WeakDelegate { get; set; }
+
+        // @required @property (readonly, assign, nonatomic) AKFLoginType loginType;
+        [Abstract]
+        [Export("loginType", ArgumentSemantic.Assign)]
+        LoginType LoginType { get; }
+    }
+
+    // @protocol AKFViewControllerDelegate <NSObject>
+    [Protocol, Model]
+    [BaseType(typeof(NSObject), Name = "AKFViewControllerDelegate")]
+    interface ViewControllerDelegate
+    {
+        // @optional -(void)viewController:(UIViewController<AKFViewController> *)viewController didCompleteLoginWithAuthorizationCode:(NSString *)code state:(NSString *)state;
+        [Export("viewController:didCompleteLoginWithAuthorizationCode:state:")]
+        void CompletedLoginWithCode(ViewController viewController, string code, string state);
```

Comment:
I believe I follow... I ended up creating the Events too because I thought that someone will probably want this functionality. Since I needed to specify a EventArgs for the additional parameters, I tried to follow how it was done in Google Mobile Ads link provided - please let me know if I followed it correctly.